src/app/lecture/knowledge-node/learning-objects/video/model/video.model.ts

```diff
@@ -0,0 +1,24 @@
+import { LearningObject } from '../../model/learning-object.model';
+import { Type } from '@angular/core';
+import { LearningObjectComponent } from '../../learning-object-component';
+import { VideoComponent } from '../video.component';
+
+export class Video extends LearningObject {
+
+  url: string;
+
+  constructor(obj?: any) {
+    if (obj) {
+      super(obj);
+      this.url = obj.url;
+    }
+  }
+
+  getVideoId(): string {
+    return this.url.split('/').pop().slice(-11);
```

Comment:
Is it always -11?

Reply:
The video ID is always 11 characters long. Although there are some other link formats this method wouldn't parse (links with additional query parameters), I couldn't find a regex that would support every youtube link format, so I decided to keep it simple by supporting default and shortened links.